contracts/sol6/KyberMatchingEngine.sol

```diff
@@ -94,14 +94,19 @@ contract KyberMatchingEngine is KyberHintHandler, IKyberMatchingEngine, Withdraw
             bytes memory unpackedHint;
             if (src == ETH_TOKEN_ADDRESS) {
                 (, unpackedHint) = unpackT2THint(hint);
-                (tradeType, reserveIds, splitValuesBps, error) = parseHint(unpackedHint);
+                (tradeType, reserveIds, splitValuesBps, error) = parseHint(dest, unpackedHint, false);
             }
             if (dest == ETH_TOKEN_ADDRESS) {
                 (unpackedHint, ) = unpackT2THint(hint);
-                (tradeType, reserveIds, splitValuesBps, error) = parseHint(unpackedHint);
+                (tradeType, reserveIds, splitValuesBps, error) = parseHint(src, unpackedHint, true);
             }
         } else {
-            (tradeType, reserveIds, splitValuesBps, error) = parseHint(hint);
+            if (src == ETH_TOKEN_ADDRESS) {
+                (tradeType, reserveIds, splitValuesBps, error) = parseHint(dest, hint, false);
```

Comment:
we decided network should do this validation.
And it shouldn't trust matching engine

seems to me now the validation will be done twice for trade and get rate
right?

Reply:
Changed to external functions only.